src/apps/material-search/useGetSelectedWork.ts

```diff
@@ -5,27 +5,78 @@ import { Work } from "../../core/utils/types/entities";
 import { WorkId } from "../../core/utils/types/ids";
 import { ManifestationMaterialType } from "../../core/utils/types/material-type";
 
+export type WorkErrorType = "work-not-found" | "material-type-not-found" | null;
+
 interface UseGetSelectedMaterialReturn {
   work: Work | null;
   selectedWorkId: WorkId | string;
   selectedMaterialType: ManifestationMaterialType | null;
   availableMaterialTypes: ManifestationMaterialType[] | null;
   isSelectedWorkLoading: boolean;
+  workError: WorkErrorType;
```

Comment:
Consider updating this to two different values: `isWorkError` and `workError`. 

As I see it this would make the hook feel more like [what React Query does elsewhere](:+1: LGTM

There are some parts of the hooks that get complex/confusing that you may want to to look into).